test/mockito_test.dart

```diff
@@ -352,6 +381,210 @@ void main() {
+        mock.methodWithoutArgs();
+
+        await untilCalled(mock.methodWithoutArgs());
+
+        verify(mock.methodWithoutArgs()).called(1);
+      });
+
+      test("waits for method with normal args", () async {
+        mock.methodWithNormalArgs(1);
+
+        await untilCalled(mock.methodWithNormalArgs(any));
+
+        verify(mock.methodWithNormalArgs(any)).called(1);
+      });
+
+      test("waits for method with list args", () async {
+        mock.methodWithListArgs([1]);
+
+        await untilCalled(mock.methodWithListArgs(any));
+
+        verify(mock.methodWithListArgs(any)).called(1);
+      });
+
+      test("waits for method with positional args", () async {
+        mock.methodWithPositionalArgs(1, 2);
+
+        await untilCalled(mock.methodWithPositionalArgs(any, any));
+
+        verify(mock.methodWithPositionalArgs(any, any)).called(1);
+      });
+
+      test("waits for method with named args", () async {
+        mock.methodWithNamedArgs(1, y: 2);
+
+        await untilCalled(mock.methodWithNamedArgs(any, y: any));
+
+        verify(mock.methodWithNamedArgs(any, y: any)).called(1);
+      });
+
+      test("waits for method with two named args", () async {
+        mock.methodWithTwoNamedArgs(1, y: 2, z: 3);
+
+        await untilCalled(mock.methodWithTwoNamedArgs(any, y: any, z: any));
+
+        verify(mock.methodWithTwoNamedArgs(any, y: any, z: any)).called(1);
+      });
+
+      test("waits for method with obj args", () async {
+        mock.methodWithObjArgs(new RealClass());
+
+        await untilCalled(mock.methodWithObjArgs(any));
+
+        verify(mock.methodWithObjArgs(any)).called(1);
+      });
+
+      test("waits for function with positional parameters", () async {
+        mock.typeParameterizedFn([1, 2], [3, 4], [5, 6], [7, 8]);
+
+        await untilCalled(mock.typeParameterizedFn(any, any, any, any));
+
+        verify(mock.typeParameterizedFn(any, any, any, any)).called(1);
+      });
+
+      test("waits for function with named parameters", () async {
+        mock.typeParameterizedNamedFn([1, 2], [3, 4], y: [5, 6], z: [7, 8]);
+
+        await untilCalled(
+            mock.typeParameterizedNamedFn(any, any, y: any, z: any));
+
+        verify(mock.typeParameterizedNamedFn(any, any, y: any, z: any))
+            .called(1);
+      });
+
+      test("waits for getter", () async {
+        mock.getter;
+
+        await untilCalled(mock.getter);
+
+        verify(mock.getter).called(1);
+      });
+
+      test("waits for setter", () async {
+        mock.setter = "A";
+
+        await untilCalled(mock.setter = "A");
+
+        verify(mock.setter = "A").called(1);
+      });
+    });
+
+    group("on methods not yet called", () {
+      setUp(() {
+        new RealClassController(mock, streamController);
+      });
+
+      test("waits for method without args", () async {
+        streamController.add(new CallMethodsEvent());
+        verifyNever(mock.methodWithoutArgs());
```

Comment:
This is kind of tricky behavior, but I think it is valid today. This relies on the fact that `streamController.add(new CallMethodsEvent())` will call `mock.MethodWithoutArgs()` **in the next microtask** (or later, whatever). However, I think it's not actually specified whether this is the case, at the language level. E.g. ([dartpad](https://dartpad.dartlang.org/f5a0d0a940fe5778fbf6b9f04cf88fe9)

```dart
import 'dart:async';

main() async {
  var a = 1;
  new Future.value(null).then((_) => a = 2);
  print(a);
  await null;
  print(a);
}
```

prints `1, 2` today, but is not guaranteed to. All implementations today, however, _do_ execute that `a = 2` in the next microtask, so we're safe today. If Dart someday breaks this, it will be limited to these tests, not the implementation.